src/Broadway/EventStore/ConcurrencyConflictResolvingEventStore.php

```diff
@@ -94,6 +96,16 @@ private function getCurrentPlayhead(DomainEventStream $committedEvents): int
         return $playhead;
     }
 
+    private function getStartingPlayhead(DomainEventStream $uncommittedEvents): int
+    {
+        $events = iterator_to_array($uncommittedEvents);
+        /** @var DomainMessage $firstEvent */
+        $firstEvent = current($events);
+        $playhead = $firstEvent->getPlayhead();
```

Comment:
if `$uncommittedEvents` is empty, I'd find it odd to receive a `DeplicatePlayheadException`.

Anyway, I added a check to skip every work if the `DomainEventStream` is empty

Reply:
I don't mean to throw a `DuplicatePlayheadException`, I'm just saying that if `$uncommittedEvents` is empty PHP will error because the `$firstEvent->getPlayhead()` method does not exist.

Maybe we should add a test for appending zero uncommitted events to be sure.

Reply:
I meant that this private method is called only in case of a `DuplicatePlayheadException`. And in that case I would find it odd to receive an empty `$uncommittedEvents`.

Anyway, I added the test to ensure everything works as expected.

Moreover, I added a `$id = (string)$id` to throw a `Throwable` even in case `$uncommittedEvents` is empty, to avoid creating a breaking change on the API

Reply:
@othillo is there anything I could do on this to get this merged?